### ISSUE TITLE
Implement Prisma Raw Queries

### ISSUE THREAD

**Issue by gamemann** [bestserversio/bestservers#26]
This issue is to track progress on implementing the use of raw Prisma/PSQL queries to retrieve servers in the server browser. While this specific raw query is very complex, it will be needed in the future when we implement server ratings and sorting by geo-location.

**Comment by gamemann** [comment]
I've implemented a raw Prisma/PSQL query to retrieve servers inside of the server browser ([here](https://github.com/bestserversio/bestservers/blob/main/src/utils/servers/content.ts#L11)). Initially, I wanted to implement this later on alongside server ratings and geo-location based sorting due to the time required and complexity of the query, but an issue arose using the standard Prisma API with tRPC when retrieving servers in my local Best Servers setup. This issue occurred quite frequently and basically prevented the server list from loading more servers since it thought the list was completely loaded when it was not.

The issue only became noticeable when [Spy](https://github.com/bestserversio/spy) was updating servers frequently. Since we use cursor-based pagination, when the next server cursor was retrieved (online), if that server was updated by Spy to offline by the time the next batch of servers were retrieved (as you scroll down the server list), it would stop loading servers. I'm not entirely sure why this occurred, but my guess is Prisma did a lookup on the cursor where online had to be set to true, but it was set to false in this case. What's strange is I tried reproducing this issue yesterday by setting the same next cursor item to non-visible before loading more servers (there's a where clause by default that only shows visible servers). For some reason, no issues occurred in this case which is why I was stumped for a while.

Anyways, I've implemented cursor-based pagination as raw queries in the past within my [Best Mods](https://bestmods.io) website (see [this](https://github.com/bestmods/bestmods/issues/23) issue for reference) and felt like this issue shouldn't occur with raw queries since we don't need to use any where clauses on the next server cursor other than a lookup on the ID (which never changes, obviously).

After implementing the new raw query into Best Servers, I haven't had this issue yet. I believe it happened often in my local setup because a lot of servers are inaccurately set to offline due to my local Spy instance having very high server update/add limits with low timeout values resulting in many servers being set to offline when they are in reality online (false-positives). These Spy settings will be tuned and scaled on initial release so even if this issue kept occurring, it should be a lot less frequent than I experienced.

There is still a small side effect where dynamic server information such as current user counts will result in the server list being slightly out of order or some servers will be missing when sorting by the current user count for example (e.g. if the next server cursor goes from 23 users to 20 users between retrieving the next batch of servers). However, combating this issue would be difficult and the amount of processing power needed isn't really ideal in my opinion (basically we'd have to retrieve all server IDs in the full query and use that as the cursor solely while going down the list). I'm not sure if I can cache the entire query somehow just for that specific user so it doesn't update until they refresh the server browser again or change its filters. Given this isn't that big of a deal in my opinion, I'm going to leave it be until later.

I do plan on confirming raw queries fixed the bigger issue mentioned before by returning the online status of the next cursor used and printing to the console if online is false. I will post one more comment once the fix is confirmed through this method.
